src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,17 +1,121 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.IBookRepository;
 
-@Controller
+
+/**
+ * This class is the service responsible of handling the requests related to the book api
+ */
+
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private IBookRepository bookRepository;
```

Comment:
A better approach is to use **constructor dependency injection**. It's better for unit testing, because it decouples the use of Spring's context for these. Additionally, when there is only one constructor, we do not need the `@Autowired` annotation, Spring recognizes it automatically and injects the respective beans.
```suggestion
    private final BookRepository bookRepository;
    
    public BookController(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }
```